### PR TITLE
add intreehooks

### PR DESCRIPTION
**Pull request by ocefpaf**
Needed in https://github.com/conda-forge/poetry-feedstock/pull/19

There is no license file upstream. I opened an issue about it.

@takluyver do you want to be a maintainer here?

### CODE REVIEW

**Comment by ocefpaf** [comment]
I believe we can ignore the AppVeyor failure here. Also, it is a `noarch` package.

@conda-forge/staged-recipes this is ready for review.

File: recipes/intreehooks/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  sha256: 87e600d3b16b97ed219c078681260639e77ef5a17c0e0dbdd5a302f99b4e34e1
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (beckermr) [inline recipes/intreehooks/meta.yaml]: I don't think we can have noarch when the recipe uses the compiler. What am I missing?

Comment (ocefpaf) [reply recipes/intreehooks/meta.yaml]: Wait, I believe that was a mistake on my part. Let me double check the package.

File: recipes/intreehooks/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+test:
+  imports:
+    - intreehooks
+
+about:
+  home: https://github.com/takluyver/intreehooks
+  license: MIT
+  license_family: MIT
+  # https://github.com/takluyver/intreehooks/issues/1
+  # license_file: LICENSE
```
Comment (beckermr) [inline recipes/intreehooks/meta.yaml]: We need to ship the license file even if an issue is made IIUIC. Let's make sure to add it here.

Comment (ocefpaf) [reply recipes/intreehooks/meta.yaml]: A generic MIT license without the copyright header may be as problematic as a missing license, no?

Comment (beckermr) [reply recipes/intreehooks/meta.yaml]: Ack. Sorry. I misunderstood the situation. Ignore this. It is clear this is supposed to be MIT.

Comment (ocefpaf) [reply recipes/intreehooks/meta.yaml]: No problem. I'll ping the author upstream to solve this ASAP.

Comment (beckermr) [reply recipes/intreehooks/meta.yaml]: Yup. LGTM and merged!

**Comment by takluyver** [comment]
I'm hoping intreehooks won't be required for long, as it's meant to be replaced by a standardised means of doing the same thing (`backend-path`). So I don't mind being a maintainer, but I might get grumbly if it requires me to pay attention beyond the next 6 months or so. :wink:

File: recipes/intreehooks/meta.yaml (hunk 2 above)
Comment (isuruf) [reply recipes/intreehooks/meta.yaml]: In that case, we are not allowed to package this as a conda package.

Comment (beckermr) [reply recipes/intreehooks/meta.yaml]: We have packaged code like this before when it was clear what was meant in the upstream repo. If you think we should remove the package, then we can delete from master. Staged recipes conversion is broken anyways.

**Comment by ocefpaf** [comment]
> I'm hoping intreehooks won't be required for long, as it's meant to be replaced by a standardised means of doing the same thing (`backend-path`). So I don't mind being a maintainer, but I might get grumbly if it requires me to pay attention beyond the next 6 months or so.

Got it. I'm packaging this for `poetry`. BTW, do you mind helping us with it the license issue?

**Comment by takluyver** [comment]
I'm not even sure it counts as a creative work, to be honest. But I've added a LICENSE file on Github. Do you need me to make a new release as well?

File: recipes/intreehooks/meta.yaml (hunk 2 above)
Comment (isuruf) [reply recipes/intreehooks/meta.yaml]: > We have packaged code like this before when it was clear what was meant in the upstream repo.

That's just wrong. A condition of MIT license is that the copyright notice is distributed, but if there's no copyright, we just can't distribute.

**Comment by ocefpaf** [comment]
> I'm not even sure it counts as a creative work, to be honest. But I've added a LICENSE file on Github.

Thanks!

> Do you need me to make a new release as well?

No need. I can add it in the feedstock and we'll remove once you are ready for a new release.

File: recipes/intreehooks/meta.yaml (hunk 2 above)
Comment (beckermr) [reply recipes/intreehooks/meta.yaml]: OK. I had consulted with core before when doing this, but happy to not as well. Thanks!

**Pull request merged**